interpreter/runtime/data.ml

```diff
@@ -1,8 +1,42 @@
 type data = string ref
 type t = data
+type address = Memory.address
+
+exception Bounds
 
 let alloc bs = ref bs
 let size seg = I64.of_int_u (String.length !seg)
-let load seg i = (!seg).[Int64.to_int i]
-let bytes seg = !seg
 let drop seg = seg := ""
+
+let load_byte seg a =
+  try (!seg).[Int64.to_int a] with Invalid_argument _ -> raise Bounds
```

Comment:
Hi @rossberg,

`Invalid_argument` is not raised when compiling with `-unsafe` (which disables bound checking for array and string accesses). You'll either get a segfault or a wrong value. It is considered a good practice to check bounds manually when they may be wrong - because of e.g. user input.

We're currently using the reference interpreter as a reference for fuzzing our own interpreter. We would like to make it a little bit faster by using `-unsafe`. would you accept a PR switching all `try ... with Invalid_argument _` to explicit bound checks ?

Reply:
Interesting, I wasn't aware of that. Seems fishy. :)  But sure, I'm fine doing explicit bounds checks. But for consistency, I would suggest doing the same for table, elem, and data, then.